TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -73,5 +73,74 @@ class MetAPI{
                 }
             }
     }
-
+    
+    func search(parameters: [SearchParameter], completion: @escaping (SearchResponse?) -> Void) {
+        var urlString: String = self.urlBaseString
+        let urlStringSuffix: String = "public/collection/v1/search"
+        urlString.append(urlStringSuffix)
+        var searchParameters: [String : String] = self.fillingParameters(parameters: parameters)
+        self.networkManager.get(
+            urlString: urlString,
+            parameters: searchParameters) { data in
+                let jsonDecoder = JSONDecoder()
+                jsonDecoder.keyDecodingStrategy = .useDefaultKeys
+                if let data = data {
+                    do {
+                        let result = try jsonDecoder.decode(SearchResponse.self, from: data)
+                        completion(result)
+                    } catch {
+                        completion(nil)
+                    }
+                } else {
+                    completion(nil)
+                }
+            }
+    }
+    
+    private func fillingParameters (parameters: [SearchParameter]) -> [String : String] {
+        var searchParameters: [String : String] = [ : ]
+        for parameter in parameters {
+            switch parameter {
+            case .q(let text):
+                searchParameters["q"] = text
+            case .isHighlight(let bool):
```

Comment:
`bool`: bad name
please provide meaningful name for the associated values.

Reply:
fixed